File: career-planner-frontend/src/api/assessment.ts
import request from '@/utils/request'

// 测评类型接口
export interface Assessment {
  id: number
  code: string
  name: string
  description: string
  icon: string
  color: string
  duration: number
  question_count: number
  created_at: string
}

export interface QuestionOption {
  text: string
  value: string
  score: number
}

export interface AssessmentQuestion {
  id: number
  question_text: string
  question_order: number
  dimension: string | null
  options: QuestionOption[]
}

export interface AssessmentDetail extends Assessment {
  questions: AssessmentQuestion[]
}

export interface DimensionScore {
  dimension: string
  score: number
  max_score: number
  percentage: number
}

export interface AssessmentReport {
  id: number
  user_id: number
  assessment_id: number
  result_type: string
  result_name: string
  scores: Record<string, any>
  dimensions: DimensionScore[] | null
  suggestions: string | null
  total_score: number
  completed_at: string
  created_at: string
  assessment?: Assessment
}

export interface ReportListItem {
  id: number
  assessment_id: number
  assessment_name: string
  assessment_type: string
  result_type: string
  result_name: string
  completed_at: string
  color: string | null
}

export interface AnswerSubmit {
  question_id: number
  answer: string
}

export interface AssessmentSubmit {
  assessment_id: number
  answers: AnswerSubmit[]
}

// API方法

/**
 * 获取所有测评类型
 */
export function getAssessments() {
  return request.get<any, Assessment[]>('/assessment/list')
}

/**
 * 获取测评详情（包含题目）
 */
export function getAssessmentDetail(assessmentId: number) {
  return request.get<any, AssessmentDetail>(`/assessment/${assessmentId}`)
}

/**
 * 提交测评答案
 */
export function submitAssessment(data: AssessmentSubmit, userId: number) {
  return request.post<any, AssessmentReport>(`/assessment/submit?user_id=${userId}`, data)
}

/**
 * 获取我的测评报告列表
 */
export function getMyReports(userId: number) {
  return request.get<any, ReportListItem[]>(`/assessment/reports/my?user_id=${userId}`)
}

/**
 * 获取报告详情
 */
export function getReportDetail(reportId: number, userId: number) {
  return request.get<any, AssessmentReport>(`/assessment/report/${reportId}?user_id=${userId}`)
}

File: career-planner-backend/app/schemas/assessment.py
"""
职业测评相关Schema
"""
from datetime import datetime
from typing import List, Optional, Dict, Any
from pydantic import BaseModel


# ==================== 测评题目 ====================
class QuestionOption(BaseModel):
    """题目选项"""
    text: str
    value: str
    score: int = 0


class AssessmentQuestionBase(BaseModel):
    """测评题目基础"""
    question_text: str
    question_order: int
    dimension: Optional[str] = None
    options: List[QuestionOption]


class AssessmentQuestionResponse(AssessmentQuestionBase):
    """测评题目响应"""
    id: int

    class Config:
        from_attributes = True


# ==================== 测评类型 ====================
class AssessmentBase(BaseModel):
    """测评基础"""
    code: str
    name: str
    description: Optional[str] = None
    icon: Optional[str] = None
    color: Optional[str] = None
    duration: Optional[int] = None
    question_count: Optional[int] = None


class AssessmentResponse(AssessmentBase):
    """测评响应"""
    id: int
    created_at: datetime

    class Config:
        from_attributes = True


class AssessmentDetailResponse(AssessmentResponse):
    """测评详情响应（包含题目）"""
    questions: List[AssessmentQuestionResponse] = []


# ==================== 测评答案提交 ====================
class AnswerSubmit(BaseModel):
    """单个答案提交"""
    question_id: int
    answer: str  # 选择的选项value


class AssessmentSubmit(BaseModel):
    """测评提交"""
    assessment_id: int
    answers: List[AnswerSubmit]


# ==================== 测评报告 ====================
class DimensionScore(BaseModel):
    """维度得分"""
    dimension: str
    score: int
    max_score: int
    percentage: float


class AssessmentReportBase(BaseModel):
    """测评报告基础"""
    assessment_id: int
    result_type: str
    result_name: str
    scores: Dict[str, Any]
    dimensions: Optional[List[DimensionScore]] = None
    suggestions: Optional[str] = None
    total_score: float = 0


class AssessmentReportResponse(AssessmentReportBase):
    """测评报告响应"""
    id: int
    user_id: int
    completed_at: datetime
    created_at: datetime

    class Config:
        from_attributes = True


class AssessmentReportDetailResponse(AssessmentReportResponse):
    """测评报告详情"""
    assessment: AssessmentResponse


# ==================== 历史报告列表 ====================
class ReportListItem(BaseModel):
    """报告列表项"""
    id: int
    assessment_id: int
    assessment_name: str
    assessment_type: str
    result_type: str
    result_name: str
    completed_at: datetime
    color: Optional[str] = None

    class Config:
        from_attributes = True
